backend/bot/index.js

```diff
@@ -14,7 +27,20 @@ bot.on('start', async () => {
   }
 });
 
-const postResultToSlack = (match) => {
+
+const ratingComparator = (a, b) => b.rating - a.rating
+
+const reportMatchOnSlack = async (match, oldUsers, newUsers) => {
+  await postMatchResult(match)
+  const oldRankings = oldUsers.sort(ratingComparator)
+  const newRankings = newUsers.sort(ratingComparator)
+  const shouldUpdatePurpose = await postRankingChangeMessage(oldRankings, newRankings)
```

Comment:
await postRankingChangeMessage(oldRankings, newRankings)
const shouldUpdatePurpose = hasLeaderboardChanged(oldRankings, newRankings)
if (shouldUpdatePurpose) {
  await updtePurpose(...)
}

Reply:
maybe a separate class - leaderboardManager(oldrankoings, newrankings)
with functions
- hasLeaderBoardChanged
- getLeaderBoardRankings
?
this would separate the leaderboard related logic 
let me think

Reply:
I can't tell if you're serious rn or not